apps/aecore/lib/aecore/peers/sync.ex

```diff
@@ -112,5 +147,60 @@ defmodule Aecore.Peers.Sync do
       end
     end)
   end
-end
 
+  # Builds a chain, starting from the given block,
+  # until we reach a block, of which the previous block is the highest in our chain
+  # (that means we can add this chain to ours)
+  defp build_chain(state, block, chain) do
```

Comment:
I attach my previous comment. This is only optimization so we could leave it as is, but it would be nice to change to the logic suggested in the comment.

> Aside from what I commented, I would suggest to change from build_chain function to sorting by height and then simply going through all blocks just once. The problem with current solution is: if we get a chain of blocks of length n, but we fail to get a parent block of that chain (an peer gives us a lot of unknown blocks, but no parent of those blocks) then we will call build_chain for every block in that chain and go through all of it's parents every time. So on average we will analyse the block in build_chain n/2 times.

Reply:
we discussed, not to optimize for now, we can do it at a later point in time, as this would introduce some more complexity for now

Reply:
ok